src/Expression/ForClasses/DependsOnlyOnTheseNamespaces.php

```diff
@@ -28,7 +28,7 @@ public function describe(ClassDescription $theClass, string $because): Descripti
         return new PositiveDescription("should depend only on classes in one of these namespaces: $desc", $because);
     }
 
-    public function evaluate(ClassDescription $theClass, Violations $violations, string $because): void
+    public function evaluate(ClassDescription $theClass, Violations $violations, string $because, bool $stopOnFailure): void
```

Comment:
I don't like that we need to pass the stopOnFailure to all the rules...
I have 2 ideas to solve this problem:
 - we accept that with rules like this one we accept multiple errors, because are "linked"
 - or we change completely design, and we pass the sopOnFailure flag to the Violations class, and if it is enabled violations->add throws and exception that interrupt the execution, and is catched somewhere... so this make the first violations add stop the execution without touching all the rules code.

Reply:
The second solution was my first thought, but after I decided to try to add this flag.
I can try to do the same PR with your second solution and we can understand which is the better, or if you want to create the PR you are welcome :)

Reply:
I created the PR #262 and I have mixed feelings about it...
The exception is not really exceptional, there is an empty catch... but the implementation is really simple and it's contained where you expect it...